python/cudf/cudf/core/series.py

```diff
@@ -973,17 +981,30 @@ def __repr__(self):
             output = (
                 preprocess.astype("O").fillna("null").to_pandas().__repr__()
             )
+        elif isinstance(
+            preprocess._column, cudf.core.column.CategoricalColumn
+        ):
+            min_rows = (
+                height
+                if get_option("display.max_rows") == 0
+                else get_option("display.min_rows")
+            )
+            show_dimensions = get_option("display.show_dimensions")
```

Comment:
I think you generally need to set this to False otherwise you get the preprocessed length instead of the actual length. I.E. we slim down to the minimal amount to generate correct output to send to host.

Reply:
We'll probably need this as `show_dimensions` is either `True`/`False`/`truncate`. If `show_dimensions` is False there is no `Length` in output. 

If `show_dimensions` is `True`/`truncate` we will have intermediate length of the output as preprocessed length but at line 1019 we modify the length in output string to be `self` length which will correct the output length.

```
elif lines[-1].startswith("Length: "):
                lines = lines[:-1]
                lines.append("Length: %d" % len(self))
                lines[-1] = lines[-1] + ", "
```